### DISCUSSION TITLE
Metrics and FPS

### DISCUSSION THREAD

**Discussion by Raknash3** [ultralytics/yolov5#2310] [Q&A] [answered]
How can I add more metrics apart from the ones already initialized and is it possible to display the FPS while running detect.py on video files?

**Comment by glenn-jocher** [comment]
@Raknash3 you can compute your FPS by dividing your framecount by your total time displayed at the end of detect.py.

What additional metrics are you interested in?

**Reply by Raknash3** [reply]
Thanks for replying.

I wish to add recall and precision at different thresholds ( 30%, 60% and 90%) while testing, does setting the '--conf' argument to the required values take care of this . And I have few more questions.

1. Can I use the argument '--verbose' and '--augment' in train.py ?
2.  I ran train.py on a custom dataset and in the results.png , I saw a graph with the title 'Box', is it same as GIoU?
3. While training on custom dataset can I use '--evolve' or only after I get the weights after training should I retrain with '--evolve' for hyperparameter tuning?

**Comment by glenn-jocher** [comment]
@Raknash3 test.py actually computes precision and recall at 1000 different thresholds and then presents the P and R values at the highest F1 value. This is all done here:
https://github.com/ultralytics/yolov5/blob/cbd55da5d24becbe3b94afaaa4cdd1187a512c3f/utils/metrics.py#L40-L79

GIoU loss was renamed to Box loss several months ago.

For evolution see the hyperparameter evolution tutorial:

## Tutorials

* [Hyperparameter Evolution](https://github.com/ultralytics/yolov5/issues/607)

**Comment by glenn-jocher** [comment]
Also, by default P and R are plotted for all thresholds after training completes. 

![image](https://user-images.githubusercontent.com/26833433/109369894-2c8d2b80-7853-11eb-9fdd-2f28a49fbd93.png)

![image](https://user-images.githubusercontent.com/26833433/109369903-3151df80-7853-11eb-94e2-d48d8171ba69.png)

**Comment by Raknash3** [comment]
Thanks a lot, this really helped me and what about my first question, can I use arguments like '--augment' and '--verbose' on train files. Because, I followed the TTA tutorial and you have used '--augment' only on test dataset.

**Comment by glenn-jocher** [comment, marked as answer]
@Raknash3 available arguments are shown in each file, i.e. for detect.py arguments:
https://github.com/ultralytics/yolov5/blob/cbd55da5d24becbe3b94afaaa4cdd1187a512c3f/detect.py#L147-L166

**Comment by Raknash3** [comment]
Thank you.
